lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.Model.rb

```diff
@@ -914,6 +914,225 @@ def model_set_central_preheat_coil_spm(model, thermalZones, coil)
+  # - 'flase' otherwise
+  #
+  # @author Xuechen (Jerry) Lei, PNNL
+  # @param model [OpenStudio::Model::Model] Openstudio model
+  def model_add_dcv_requirement_properties(model)
+    model.getAirLoopHVACs.each do |air_loop_hvac|
+      if user_model_air_loop_hvac_demand_control_ventilation_required?(air_loop_hvac)
+        air_loop_hvac.thermalZones.each do |thermal_zone|
+          thermal_zone.additionalProperties.setFeature('airloop dcv required by 901', true)
+
+          # the zone level dcv requirement can only be true if it is in an airloop that is required to have DCV
+          if user_model_zone_demand_control_ventilation_required?(thermal_zone)
+            thermal_zone.additionalProperties.setFeature('zone dcv required by 901', true)
+          end
+        end
+      end
+    end
+
+    # mark unmarked zones
+    model.getThermalZones.each do |zone|
+      next if zone.additionalProperties.hasFeature('airloop dcv required by 901')
+
+      zone.additionalProperties.setFeature('airloop dcv required by 901', false)
+    end
+
+    model.getThermalZones.each do |zone|
+      next if zone.additionalProperties.hasFeature('zone dcv required by 901')
+
+      zone.additionalProperties.setFeature('zone dcv required by 901', false)
+    end
+  end
+
+  # based on previously added flag, raise error if DCV is required but not implemented in zones, in which case
+  # baseline generation will be terminated; raise warning if DCV is not required but implemented, and continue baseline
+  # generation
+  #
+  # @author Xuechen (Jerry) Lei, PNNL
+  # @param model [OpenStudio::Model::Model] Openstudio model
+  def model_raise_user_model_dcv_errors(model)
+    model.getThermalZones.each do |thermal_zone|
+      if thermal_zone.additionalProperties.getFeatureAsBoolean('zone DCV implemented in user model').get &&
+         (!thermal_zone.additionalProperties.getFeatureAsBoolean('zone dcv required by 901').get ||
+           !thermal_zone.additionalProperties.getFeatureAsBoolean('airloop dcv required by 901').get)
+        OpenStudio.logFree(OpenStudio::Warn, 'openstudio.standards.Model', "For thermal zone #{thermal_zone.name}, ASHRAE 90.1 2019 6.4.3.8 does NOT require this zone to have demand control ventilation, but it was implemented in the user model, Appendix G baseline generation will continue!")
+      end
+      if thermal_zone.additionalProperties.getFeatureAsBoolean('zone dcv required by 901').get &&
+         thermal_zone.additionalProperties.getFeatureAsBoolean('airloop dcv required by 901').get &&
+         !thermal_zone.additionalProperties.getFeatureAsBoolean('zone DCV implemented in user model').get
+        OpenStudio.logFree(OpenStudio::Error, 'openstudio.standards.Model', "For thermal zone #{thermal_zone.name}, ASHRAE 90.1 2019 6.4.3.8 requires this zone to have demand control ventilation, but it was not implemented in the user model, Appendix G baseline generation should be terminated!")
```

Comment:
For this test, we should report the error, but not halt the generation. We should report to the special PRM log file using 'prm.log' instead of 'openstudio.standards.Model'. You can pull Appendix_G_Dev to get that capability into your branch. I think we should have a standard prefix for cases where the user model does not meet mandatory requirements.  Like start with: 'Mandatory Requirement Not Met:  '. Then the user can filter or sort the prm.log file to see those issues together.  I don't think the phrase "Appendix G baseline generation should be terminated" is meaningful if we are not actually terminating.